### PR TITLE
11.0 cost report pending invoices

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web#id=9016&view_type=form&model=project.task&action=292&active_id=413&menu_id=200

### CODE REVIEW

File: project_cost_report/report.py [hunk 1]
```diff
@@ -437,3 +439,100 @@ def get_foldable_categories(self, project_id):
...
+
+        :param project: the project.project record
+        :rtype: purchase.order.line
+        """
+        domain = [
+            ('account_analytic_id', '=', project.analytic_account_id.id),
+            ('order_id.state', 'in', ('purchase', 'done')),
+        ]
+        lines = self.env['purchase.order.line'].search(domain)
+        return lines.filtered(lambda l: purchase_line_is_pending_invoice(l))
```
Comment (ddufresne) [inline project_cost_report/report.py]: Codacy Issue found: [Lambda may not be necessary](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3730559)

File: project_cost_report/static/src/js/report_action.js [hunk 2]
```diff
@@ -370,6 +372,23 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
+     *
+     * The form view of the PO is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    purchaseOrderNameClicked(event){
+        event.preventDefault();
+        var orderId = getRecordIdFromEvent(event);
+        this.do_action({
+            res_model: "purchase.order",
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'res_model' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3730559)

File: project_cost_report/static/src/js/report_action.js [hunk 3]
```diff
@@ -370,6 +372,23 @@ var ReportAction = Widget.extend(ControlPanelMixin, {
...
+            views: [[false, "form"]],
+            type: "ir.actions.act_window",
+            res_id: orderId,
```
Comment (ddufresne) [inline project_cost_report/static/src/js/report_action.js]: Codacy Issue found: [Identifier 'res_id' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3730559)

File: project_cost_report/report.py [hunk 4]
```diff
@@ -437,3 +439,100 @@ def get_foldable_categories(self, project_id):
...
+    """Represents a purchase order in the cost report.
+
+    This object is used to display the share of amount pending invoice
+    that is bound to the current project.
+
+    :param order: the purchase order represented.
+    :param project: the project for which to render the report.
+    """
+
+    def __init__(self, order: models.Model, project: models.Model):
```
Comment (foutoucour) [inline project_cost_report/report.py]: pourquoi un objet ici?
Ça ressemble à une fonction qui retourne la valeur de `total` non ?

**Review by foutoucour — CHANGES_REQUESTED** [review]

File: project_cost_report/report.py (hunk 4 above)
Comment (ddufresne) [reply project_cost_report/report.py]: Oui, je suis d'accord.
Au départ, j'avais prévu me servir de cette objet pour contenir d'autres variables. Là, c'est overkill d'utiliser un objet.

**Comment by ddufresne** [comment]
@foutoucour fixed

File: project_cost_report/report.py [hunk 5]
```diff
@@ -437,3 +439,96 @@ def get_foldable_categories(self, project_id):
...
+
+        :param project: the project.project record
+        :rtype: purchase.order.line
+        """
+        domain = [
+            ('account_analytic_id', '=', project.analytic_account_id.id),
+            ('order_id.state', 'in', ('purchase', 'done')),
+        ]
+        lines = self.env['purchase.order.line'].search(domain)
+        return lines.filtered(lambda l: purchase_line_is_waiting_invoice(l))
```
Comment (ddufresne) [inline project_cost_report/report.py]: Codacy Issue found: [Lambda may not be necessary](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3730559)

**Review by foutoucour — APPROVED** [review]

**Pull request merged**